Create logging functionality

Each build need to be logged in a persistent way. One way to do this would be to write the output from the test framework and type checker to a json file. 

Instructions from canvas: 
Property (CI feature): the CI server keeps the history of the past builds. This history persists even if the server is rebooted. Each build is given a unique URL, that is accessible to get the build information (commit identifier, build date, build logs). One URL exists to list all builds.

Preparation: the students document the build list URL in the README.

Assessment: The grader opens the build list URL by clicking on it in the README. The grader randomly clicks on a build and assesses the appropriateness of the information.

If we are to log the outcome of checks that are run against the code we are going to understand the output of the linting software. Example output from pylint is: 
************* Module ci_server
ci_server\__init__.py:1:0: C0114: Missing module docstring (missing-module-docstring)
************* Module ci_server.server
ci_server\server.py:20:9: W0511: TODO: implement logic for serving files here (fixme)
ci_server\server.py:23:9: W0511: TODO: implement CI logic here (fixme)
ci_server\server.py:1:0: C0114: Missing module docstring (missing-module-docstring)
ci_server\server.py:9:4: C0103: Method name "do_GET" doesn't conform to snake_case naming style (invalid-name)
ci_server\server.py:9:4: C0116: Missing function or method docstring (missing-function-docstring)
ci_server\server.py:15:8: W0702: No exception type(s) specified (bare-except)
ci_server\server.py:13:19: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
ci_server\server.py:13:19: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
ci_server\server.py:11:12: W0201: Attribute 'path' defined outside __init__ (attribute-defined-outside-init)
************* Module run
run.py:2:0: C0411: standard import "from http.server import HTTPServer" should be placed before "from ci_server import CIServer" (wrong-import-order)
************* Module test
tests\ci_server\test.py:1:1: W0511: TODO: implement new tests (fixme)

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,5 @@
 venv/
 past_builds/
 __pycache__
+buildData.dat
+Build*.txt

--- a/ci_server/server.py
+++ b/ci_server/server.py
@@ -1,5 +1,34 @@
 from http.server import BaseHTTPRequestHandler
 
+import time
+
+# The purpose of this function is to generate a unique identifier that serves
+# as the build log name.
+# Note: This would fail if the number of total builds would reach maxint.
+def make_log_title():
+    tob = time.localtime()
+    bString = "X"
+    bDatPath = "../logfiles/buildData.dat"
+    newBuildNum = -1
+
+    # Format the new build number.
+    with open(bDatPath) as f:
+        newBuildNum = int(f.read()[:-1])+1
+
+    # Generate a build log string.
+    bString = f"Build_{newBuildNum}_{tob.tm_year}_{tob.tm_mon}_{tob.tm_mday}_{tob.tm_hour}.txt"
+
+    with open(bDatPath, "w") as f:
+        f.truncate(0)
+        f.write(str(newBuildNum)+'\n')
+
+    return bString
+
+# The purpose of this function is to log the output of the tests and the linting.
+# Creates a new .txt file with the output of the linter and the tests.
+def make_log(lint_output, pytest_output):
+    with open(f"../logfiles/{make_log_title()}", "w") as f:
+        f.write(f"=== LINT OUTPUT ===\n{lint_output}\n\n=== PYTEST OUTPUT ===\n{pytest_output}\n")
 
 class CIServer(BaseHTTPRequestHandler):
     """

